platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateRoot.java

```diff
@@ -914,32 +922,18 @@ public SwirldState getSwirldState() {
      */
     @NonNull
     @Override
-    public PlatformState getPlatformState() {
-        return getChild(PLATFORM_STATE_INDEX);
+    public PlatformStateAccessor getPlatformStateAccessor() {
+        return platformState;
     }
 
     /**
-     * Set the platform state. This method shifts existing children to the right to insert
-     * {@link PlatformState} instance at the {@code PLATFORM_STATE_INDEX} index.
-     * If the platform state is already present, it doesn't do the shifting.
+     * Updates the platform state with the values from the provided instance of {@link PlatformStateAccessor}
+     *
+     * @param source a source of values
      */
     @Override
-    public void setPlatformState(@NonNull final PlatformState platformState) {
-        if (getChild(PLATFORM_STATE_INDEX) instanceof PlatformState) {
-            setChild(PLATFORM_STATE_INDEX, platformState);
-            return;
-        }
-
-        INDEX_LOOKUP.clear();
-        // Shift all existing children to the right
-        if (getNumberOfChildren() != 0) {
-            for (int i = getNumberOfChildren(); i > 0; i--) {
-                MerkleNode child = getChild(i - 1);
-                setChild(i, child.copy());
-                setChild(i - 1, null);
-            }
-        }
-        setChild(PLATFORM_STATE_INDEX, platformState);
+    public void updatePlatformStateAccessor(@NonNull final PlatformStateAccessor source) {
```

Comment:
Method name looks confusing to me. It doesn't update the accessor, it updates accessor state, did I get it right? So a better name would be `updatePlatformState()`

Reply:
Fixed